src/ipvs/ip_vs_proto_icmp.c

```diff
@@ -132,6 +132,29 @@ static bool is_icmp_reply(uint8_t type)
 		return false;
 }
 
+bool dp_vs_conn_icmp_fill_param(const struct dp_vs_iphdr *iph,
+        struct dp_vs_dest *dest, struct rte_mbuf *mbuf, 
+        bool reverse, struct dp_vs_conn_param *param)
+{
+    struct icmphdr *ich, _icmph;
+    uint16_t ports[2]; /* sport, dport */
+    ich = mbuf_header_pointer(mbuf, iph->len, sizeof(_icmph), &_icmph);
+    if (!ich)
+        return false;
+
+    ports[0] = icmp4_id(ich);
+    ports[1] = ich->type << 8 | ich->code;
+
+    /* ID may confict for diff host, should we use ID pool ? */
+    if (reverse) {
+        dp_vs_conn_fill_param(iph->af, iph->proto, &iph->daddr, &dest->addr, 
+                ports[1], ports[0], 0, param);
+    } else {
+        dp_vs_conn_fill_param(iph->af, iph->proto, &iph->daddr, &dest->addr,
```

Comment:
It was `iph->saddr`, `iph->daddr` for `<CIP, VIP>`, here use `dest->addr` for VIP, it is not correct.
```
            dp_vs_conn_fill_param(iph->af, iph->proto,
                                  &iph->saddr, &iph->daddr,
                                  ports[0], ports[1], 0, &param);
```
Since it's not easy to eliminate "duplicated" codes, suggest not combine the codes with icmp_fill_param.